### DISCUSSION TITLE
How to not wipe app folder?

### DISCUSSION THREAD

**Discussion by eidolonFIRE** [leancodepl/patrol#1706] [Q&A]
I've just upgraded to v2 and started using patrol (properly?) with `patrol test`.
However, when I do this, it wipes all the app data which is extremely inconvenient for reasons not worth going into.

I tried changing the package name in `MainActivityTest.java` but that just broke things and I'm not familiar enough with native Android env to fix it.

**Is there a way to run `patrol test` under a different app name to preserve my app data?**

**Comment by bartekpacia** [comment]
`patrol test` uninstalls the app by default before running it. To disable this behavior, use the `--no-uninstall` flag.

If that's not the cause, then make sure you _don't have_ `clearPackageData: true` in your `app/build.gradle`.

**Reply by eidolonFIRE** [reply]
I tried the `--no-uninstall` option, but I think this leaves my target device in a bad state.
I'm using vsCode to run with debugger and I get this output when trying to run the app after `patrol test --no-uninstall`. It makes me think the test app has overwritten the main app, and vsCode hasn't tried to re-install it.

```
I/com.example(13686): Compiler allocated 4533KB to compile void android.view.ViewRootImpl.performTraversals()
I/flutter (13686): Patrol (native): initialize() started
I/flutter (13686): Patrol (native): initialize() failed
E/flutter (13686): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Patrol action failed: GrpcError: initialize() failed with code UNAVAILABLE (Error connecting: SocketException: Connection refused (OS Error: Connection refused, errno = 111), address = localhost, port = 38918)
E/flutter (13686): #0      NativeAutomator._wrapRequest
native_automator.dart:237
E/flutter (13686): <asynchronous suspension>
E/flutter (13686): #1      NativeAutomator.initialize
native_automator.dart:254
E/flutter (13686): <asynchronous suspension>
E/flutter (13686): #2      main
test_bundle.dart:54
E/flutter (13686): <asynchronous suspension>
E/flutter (13686):
Application finished.
Exited (sigterm)
```
